src/global/helpers/native-date-adapter.ts

```diff
@@ -194,7 +194,7 @@ export class NativeDateAdapter implements DateAdapter<Date> {
   /** Returns the right format for the `valueAsDate` property. */
   public formatValueAsDate(value: string): Date {
     const values = value?.split('.');
-    if (!values || values.length <= 2 || values.some((v) => v === '')) {
+    if (!values || values.length <= 2 || values.some((v) => v === '' || v === '00')) {
```

Comment:
Why is this necessary? I don't yet understand why we check for '00' but not e.g. '0'.
This is not a problem, I would just like to understand it.

Reply:
This method is called after the input value has been formatted (that's why we split using the "." and not other possible separators), hence any 0 is already padded to 00. If you try to input 12.0.2012, it will automatically change to 12.00.2012 and then be marked as wrong

Reply:
I forgot to mention that we need to include this case as fault scenario because the Date constructor does allow 0s (and maybe also negative numbers), it just "overflows" like 12.00.2012 = 12.12.2011

Reply:
Ah, I see
However, `11.07.00` is technically a valid date (although it gets translated to `11.07.1900`).
Due to this, I think we should leave it as it is for now (even, if it is weird).
Or what do you think?

Reply:
Dscussed via Teams; Day and month should be validated against '00' and '0', but the year should be left as is.